src/main/scala/com/lightbend/conductr/sbt/ConductRPlugin.scala

```diff
@@ -45,26 +51,125 @@ object ConductrPlugin extends AutoPlugin {
 
   private final val TypesafePropertiesName = "typesafe.properties"
   private final val SandboxRunArgsAttrKey = AttributeKey[SandboxRunArgs]("conductr-sandbox-run-args")
   private final val sandboxRunTaskInternal = taskKey[Unit]("Internal Helper to call sandbox run task.")
 
+  private def installationScriptContentsTask: Def.Initialize[Task[Seq[String]]] = Def.taskDyn {
+    val installDir = (target in LocalRootProject).value.toPath
+
+    val rootProjectStructure = Project.extract(state.value).structure
+    val allProjects = rootProjectStructure.allProjects
+    val bundleInstallationScriptsTasks = allProjects.map { project =>
+      val projectRef = ProjectRef(rootProjectStructure.root, project.id)
+
+      val bundleConfiguration = project.configurations.find(c => c.name == Bundle.name)
+      val bundleConfigurationConfiguration = project.configurations.find(c => c.name == BundleConfiguration.name)
+
+      (bundleConfiguration, bundleConfigurationConfiguration) match {
+        case (Some(b), Some(bc)) =>
+          Def.taskDyn {
+            def formatScript(bundleName: String, bundleIdEnv: String, bundleFile: Path, bundleConfigFile: Option[Path]): String =
+              s"""
+                 |echo "Deploying $bundleName..."
+                 |$bundleIdEnv=$$(conduct load $bundleFile ${bundleConfigFile.fold("")(_ + " ")}--long-ids -q)
+                 |conduct run $${$bundleIdEnv} --no-wait -q
+                 """.stripMargin
+
+            val bundleName = (normalizedName in b in projectRef).value
+            val bundleIdEnv = bundleName.replaceAll("\\W", "_").toUpperCase + "_BUNDLE_ID"
+
+            val bundleFile = installDir.relativize((dist in b in projectRef).value.toPath)
+            val bundleConfigContents = (BundleKeys.bundleConf in bc in projectRef).value
+            val srcDir = (sourceDirectory in bc in projectRef).value / (BundleKeys.configurationName in bc in projectRef).value
+            if (bundleConfigContents.nonEmpty || srcDir.exists()) {
+              Def.task {
+                val bundleConfigFile = installDir.relativize((dist in bc in projectRef).value.toPath)
+                formatScript(bundleName, bundleIdEnv, bundleFile, Some(bundleConfigFile))
+              }
+            } else {
+              Def.task {
+                formatScript(bundleName, bundleIdEnv, bundleFile, None)
+              }
+            }
+          }
+        case _ =>
+          Def.taskDyn(Def.task(""))
+      }
+    }
+
+    def fold(a: Seq[String])(tasks: Seq[Def.Initialize[Task[String]]]): Def.Initialize[Task[Seq[String]]] =
+      tasks match {
+        case Nil     => Def.task(a)
+        case x :: xs => Def.taskDyn(fold(a :+ x.value)(xs))
+      }
+
+    fold(List.empty)(bundleInstallationScriptsTasks)
+  }
+
+  private def generateInstallationScriptTask(): Def.Initialize[Task[File]] = Def.task {
+    val installationScript = (target in LocalRootProject).value / "install.sh"
```

Comment:
I’m eradicating the Unix specific stuff now.

> On 25 May 2016, at 3:49 PM, Mirco Dotta <notifications@github.com> wrote:
> 
> In src/main/scala/com/lightbend/conductr/sbt/ConductRPlugin.scala <https://github.com/typesafehub/sbt-conductr/pull/156#discussion_r64518331>:
> 
> > +        case _ =>
> > +          Def.taskDyn(Def.task(""))
> > +      }
> > +    }
> > +
> > +    def fold(a: Seq[String])(tasks: Seq[Def.Initialize[Task[String]]]): Def.Initialize[Task[Seq[String]]] =
> > +      tasks match {
> > +        case Nil     => Def.task(a)
> > +        case x :: xs => Def.taskDyn(fold(a :+ x.value)(xs))
> > +      }
> > +
> > +    fold(List.empty)(bundleInstallationScriptsTasks)
> > +  }
> > +
> > +  private def generateInstallationScriptTask(): Def.Initialize[Task[File]] = Def.task {
> > +    val installationScript = (target in LocalRootProject).value / "install.sh"
> Might be, not sure what's the market share of Windows 10.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/typesafehub/sbt-conductr/pull/156/files/f17e4d13bd5905f96593d5698f35018270b7a388#r64518331>